### PR TITLE
autocomplete_using_suggester: improve sorting by length

### PR DESCRIPTION
**Pull request by mickeyn**
sorting by length in the query to correct current results set.

### CODE REVIEW

File: lib/MetaCPAN/Document/File/Set.pm [hunk 1]
```diff
@@ -499,28 +499,47 @@ sub autocomplete_using_suggester {
         }
         );
 
-    my @docs
-        = map { $_->{text} } @{ $suggestions->{documentation}[0]{options} };
+    my %docs;
+    for my $d ( @{ $suggestions->{documentation}[0]{options} } ) {
```
Comment (oalders) [inline lib/MetaCPAN/Document/File/Set.pm]: What is `$d`?  It's not immediately clear to me.  Maybe a more descriptive name?

**Comment by mickeyn** [comment]
this change is for the not yet used suggester based autocomplete (for testing and comparison)

File: lib/MetaCPAN/Document/File/Set.pm [hunk 2]
```diff
@@ -499,28 +499,47 @@ sub autocomplete_using_suggester {
...
+        next if exists $docs{ $d->{text} };
+        $docs{ $d->{text} } = $d->{score};
+    }
+
+    my @docs_uniq = sort { $docs{$b} <=> $docs{$a} } keys %docs;
```
Comment (oalders) [inline lib/MetaCPAN/Document/File/Set.pm]: Out of curiosity, why do these need to be sorted if it's being passed to Es in a query?

Comment (ranguard) [inline lib/MetaCPAN/Document/File/Set.pm]: ES does clever analysis, but can't order alphabetically on a suggest field!

Comment (oalders) [inline lib/MetaCPAN/Document/File/Set.pm]: Right, but it's being sorted on the way in rather than on the way out?

Comment (mickeyn) [inline lib/MetaCPAN/Document/File/Set.pm]: that's a redundant leftover from before adding the script ... will remove
I was looking for the reason it was getting low-quality results, and thought the `size => 10` which is clearly why we got random 10 matches from 'all' doc matches might be affected by the order... not smart :)

File: lib/MetaCPAN/Document/File/Set.pm (hunk 1 above)
Comment (mickeyn) [inline lib/MetaCPAN/Document/File/Set.pm]: will rename

**Pull request merged**